prompt_toolkit/layout/utils.py

```diff
@@ -1,5 +1,7 @@
 from typing import TYPE_CHECKING, Iterable, List, TypeVar, Union, cast, overload
 
+from typing_extensions import SupportsIndex
```

Comment:
``typing_extensions`` is not a required dependency, so this import should be an a ``if TYPE_CHECKING:`` block.

Reply:
I see, do you have plans to drop usage of python 3.6 soon as it is already End-Of-Life? With [pyupgrade](https://github.com/asottile/pyupgrade) one can easily migrate the whole code to use new style of annotations.